tests/components/threshold/test_binary_sensor.py

```diff
@@ -144,68 +140,42 @@ async def test_sensor_upper_hysteresis(hass: HomeAssistant) -> None:
-    hass.states.async_set("sensor.test_monitored", 12.5)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "below"
-    assert state.state == "off"
-
-    hass.states.async_set("sensor.test_monitored", 20)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "above"
+    assert state.attributes["entity_id"] == "sensor.test_monitored"
     assert state.attributes["upper"] == float(config["binary_sensor"]["upper"])
     assert state.attributes["hysteresis"] == 2.5
     assert state.attributes["type"] == "upper"
-    assert state.attributes["position"] == "above"
-    assert state.state == "on"
-
-    hass.states.async_set("sensor.test_monitored", 13)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "above"
-    assert state.state == "on"
-
-    hass.states.async_set("sensor.test_monitored", 12)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "below"
-    assert state.state == "off"
-
-    hass.states.async_set("sensor.test_monitored", 17)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "below"
-    assert state.state == "off"
 
-    hass.states.async_set("sensor.test_monitored", 18)
+    hass.states.async_set("sensor.test_monitored", to_val)
     await hass.async_block_till_done()
     state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "above"
-    assert state.state == "on"
+    assert state.attributes["position"] == expected_position
+    assert state.state == expected_state
 
-    hass.states.async_set("sensor.test_monitored", "cat")
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "unknown"
-    assert state.state == "unknown"
 
-    hass.states.async_set("sensor.test_monitored", 18)
-    await hass.async_block_till_done()
-    state = hass.states.get("binary_sensor.threshold")
-    assert state.attributes["position"] == "above"
-    assert state.state == "on"
-
-
-async def test_sensor_lower_hysteresis(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("from_val", "to_val", "expected_position", "expected_state"),
+    [
+        (None, 17.5, "above", "off"),  # threshold + hysteresis
+        (17.5, 12.5, "above", "off"),  # threshold - hysteresis
+        (12.5, 20, "above", "off"),
+        (20, 13, "above", "off"),
+        (13, 12, "below", "on"),
+        (12, 17, "below", "on"),
+        (17, 18, "above", "off"),
+        (18, "cat", "unknown", "unknown"),
+        ("cat", 18, "above", "off"),
```

Comment:
My personal preference would be to find the actual edge of the hysteresis and so with 12.4 and 17.6 as test values. But I see that you exactly recreated the test cases that were there.

Reply:
The next PR I have pending adds more test cases and fixes some edge cases. So for now I just want to refactor the tests.